sycl/test/basic_tests/reduction/known_identities.cpp

```diff
@@ -0,0 +1,120 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %t.out
+
+// Tests the existance and specializations of known identities.
+
+#include <sycl/sycl.hpp>
+
+#include <cassert>
+#include <limits>
+#include <type_traits>
+
+template <typename BinOp, typename OperandT> constexpr void checkNoIdentity() {
+  static_assert(!sycl::has_known_identity<BinOp, OperandT>::value,
+                "Operation should not have a known identity!");
+  static_assert(!sycl::has_known_identity_v<BinOp, OperandT>,
+                "Operation should not have a known identity!");
+}
+
+template <typename BinOp, typename OperandT>
+void checkIdentity(OperandT Expected) {
+  static_assert(sycl::has_known_identity<BinOp, OperandT>::value,
+                "No trait specialization for known identity!");
+  static_assert(sycl::has_known_identity_v<BinOp, OperandT>,
+                "No trait specialization for known identity!");
+  assert((sycl::known_identity<BinOp, OperandT>::value == Expected) &&
+         "Identity does not match expected.");
+  assert((sycl::known_identity_v<BinOp, OperandT> == Expected) &&
+         "Identity does not match expected.");
+}
+
+template <typename OperandT> constexpr void checkAll() {
+  if constexpr (std::is_arithmetic_v<OperandT> ||
+                std::is_same_v<std::remove_cv_t<OperandT>, sycl::half>) {
```

Comment:
I think this is too close to what the implementation does so there is little benefit of having a tests like this (better than nothing still, I guess).

A slightly different approach would be to do something like

```
check_supported(int, {std::plus<int>, 0}, 
                     {std::plus<>, 0},
                     {std::bit_and<>, -1},
                     ....);
```

I'm not sure if we can implement something like this without too much magic...

Reply:
> I think this is too close to what the implementation does so there is little benefit of having a tests like this (better than nothing still, I guess).

A benefit of having a test like this is to ensure that we do not unintentionally regress in the behavior without noticing. We could definitely be more verbose, but the test would become huge and in the end I doubt there's much benefit to that either.